go/pkg/pwapi/api_team-send-invite.go

```diff
@@ -3,14 +3,102 @@ package pwapi
 import (
 	"context"
 
+	"github.com/jinzhu/gorm"
 	"pathwar.land/pathwar/v2/go/pkg/errcode"
+	"pathwar.land/pathwar/v2/go/pkg/pwdb"
 )
 
 func (svc *service) TeamSendInvite(ctx context.Context, in *TeamSendInvite_Input) (*TeamSendInvite_Output, error) {
-	if in == nil || in.TeamID == 0 || in.UserID == 0 {
+	if in == nil || in.TeamID == "" || in.UserID == "" {
 		return nil, errcode.ErrMissingInput
 	}
 
-	ret := TeamSendInvite_Output{}
-	return &ret, errcode.ErrNotImplemented
+	userID, err := userIDFromContext(ctx, svc.db)
+	if err != nil {
+		return nil, errcode.ErrUnauthenticated.Wrap(err)
+	}
+
+	teamID, err := pwdb.GetIDBySlugAndKind(svc.db, in.UserID, "team")
+	if err != nil {
+		return nil, errcode.TODO.Wrap(err)
+	}
+
+	invitedUserID, err := pwdb.GetIDBySlugAndKind(svc.db, in.UserID, "user")
+	if err != nil {
+		return nil, errcode.TODO.Wrap(err)
+	}
+
+	// check team status
+	var team pwdb.Team
+	err = svc.db.
+		Where(pwdb.Team{
+			ID:             teamID,
+			DeletionStatus: pwdb.DeletionStatus_Active,
+		}).
+		First(&team).
+		Error
+	if err != nil {
+		return nil, errcode.ErrTeamNotExists.Wrap(err)
+	}
+
+	// check that the user is owner of the team
+	var teamOwner pwdb.TeamMember
+	err = svc.db.
+		Where(pwdb.TeamMember{
+			UserID: userID,
+			TeamID: teamID,
+			Role:   pwdb.TeamMember_Owner,
+		}).
+		First(&teamOwner).
+		Error
+	if err != nil {
+		return nil, errcode.ErrNotTeamOwner.Wrap(err)
+	}
+
+	// check that invited user is not already part of the team
```

Comment:
here you should instead check if the user is not already in a team in the same session (which includes your check btw)